design-documents/graph-ql/coverage/catalog/catalog-product-common-types.graphqls

```diff
@@ -0,0 +1,16 @@
+
+type ProductUrl {
```

Comment:
Where these types are planned to be reused outside of products compare?

Reply:
It should be aligned with the product schema

Reply:
Product schema already exists. So looks like we should not introduce new types here.

Reply:
As a consumer of this schema, I agree with @paliarush and I'd also urge you to consider adding additional interfaces where necessary, instead of special-purpose wrapper types.

Reply:
item in compare list not a product in an understanding of a product query that we have, actually it has a different much more limited set of fields. The compare list rather report then entity. it has some product attributes true. Another question is performance, I do not want to use the product as supertype everywhere where this makes sense or does not make.  we had this with Magento, as a result huge performance degradation each time as customization added or one additional available field requested. My strong belief that API should follow to business scenario requirement and provide exactly what needed for the particular scenario on a page.

Reply:
this is a good question, right now product schema has a canonical_url which is just a string, no attributes can be added. which potentially could be a problem, are we good with URL as a string, or we can add a new object field in a backward-compatible manner?

Reply:
A conformant URL is as good as an object. If we need additional properties on the URL string, there are ways to add them within the URI spec.

Reply:
fixed